src/geojson/geojson.js

```diff
@@ -0,0 +1,494 @@
+            let th = document.createElement("th");
+            let td = document.createElement("td");
+            th.appendChild(document.createTextNode(key));
+            td.appendChild(document.createTextNode(json[key]));
+            th.setAttribute("scope", "row");
+            td.setAttribute("itemprop", key);
+            row.appendChild(th);
+            row.appendChild(td);
+        }
+    }
+    return table;
+}
+
+// Takes a GeoJSON geojson and an options Object which returns a <layer-> Element
+// The options object can contain the following:
+//      label            - String, contains the layer name, if included overrides the default label mapping
+//      projection       - String, contains the projection of the layer (OSMTILE, WGS84, CBMTILE, APSTILE), defaults to OSMTILE
+//      caption          - Function | String, function accepts one argument being the feature object which produces the featurecaption string OR a string that is the name of the property that will be mapped to featurecaption
+//      properties       - Function | String | HTMLElement, a function which maps the geojson feature properties to an HTMLElement or a string that will be parsed as an HTMLElement or an HTMLElement
+//      geometryFunction - Function, A function you supply that can add classes, hyperlinks and spans to the created <map-geometry> element, default would be the plain map-geometry element
+// geojson2mapml: geojson Object <layer-> -> <layer->
+function geojson2mapml(json, options = {}, layer = null) {
+    let defaults = {
+        label: null,
+        projection: "OSMTILE",
+        caption: null,
+        properties: null,
+        geometryFunction: null
+    };
+    // assign default values for undefined options
+    options = Object.assign({}, defaults, options) ;
+
+    // If string json is received
+    if (typeof json === "string") {
+        json = JSON.parse(json);
+    }
+    let geometryType = ["POINT", "LINESTRING", "POLYGON", "MULTIPOINT", "MULTILINESTRING", "MULTIPOLYGON", "GEOMETRYCOLLECTION"];
+    let jsonType = json.type.toUpperCase();
+    let out = "";
+
+    // HTML parser
+    let parser = new DOMParser();
+
+    // initializing layer
+    if (layer === null) {
+        // creating an empty mapml layer
+        let xmlStringLayer = "<layer- label='' checked><map-meta name='projection' content='" + options.projection + "'></map-meta><map-meta name='cs' content='gcrs'></map-meta></layer->";
+        layer = parser.parseFromString(xmlStringLayer, "text/html");
+        //console.log(layer)
+        if (options.label !== null) {
+            layer.querySelector("layer-").setAttribute("label", options.label);
+        } else if (json.name) {
+            layer.querySelector("layer-").setAttribute("label", json.name);
+        } else if (json.title) {
+            layer.querySelector("layer-").setAttribute("label", json.title);
+        } else{
+            layer.querySelector("layer-").setAttribute("label", "Layer");

```

Comment:
This could be an internationalization opportunity / need, _perhaps_.  Of course the mapml-extension carries all these strings.